tests/tests/test_compat.py

```diff
@@ -34,56 +34,80 @@
 container_factory = factory.get_factory()
 
 TIMEOUT = timedelta(minutes=5)
+"""
+ COMPAT_MENDER_VERSIONS array stores the client version we test against.
+ In order to add a new client: add an entry in here and do not forget to 
+create a composition file in extra/integration-testing/docker-compose.compat-VERSION.yml
+where VERSION is the string you added here.
+"""
+COMPAT_MENDER_VERSIONS = [
+    "2.0",
+    "2.1",
+    "2.2",
+    "2.3",
+    "2.4",
+    "2.5",
+    "2.6",
+    "3.0",
+    "3.1",
+    "mender-3.2",
+    "mender-3.3",
+    "mender-3.4",
+]
 
 
 @pytest.fixture(scope="function")
-def setup_os_compat(request):
-    env = container_factory.get_compatibility_setup()
-    request.addfinalizer(env.teardown)
-    env.setup()
+def setup_os_compat():
+    def create(tag):
+        env = container_factory.get_compatibility_setup(tag=tag)
+        env.setup()
 
-    env.user = create_user(
-        "test@mender.io", "correcthorse", containers_namespace=env.name
-    )
-    env.populate_clients()
+        env.user = create_user(
+            "test@mender.io", "correcthorse", containers_namespace=env.name
+        )
+        env.populate_clients()
+
+        clients = env.get_mender_clients()
+        assert len(clients) > 0, "Failed to setup clients"
+        env.devices = []
+        for client in clients:
+            dev = MenderDevice(client)
+            dev.ssh_is_opened()
+            env.devices.append(dev)
```

Comment:
Should this expect only one client? I would make this strict with something like:

```suggestion
        clients = env.get_mender_clients()
        assert len(clients) == 1, "Failed to setup clients"
        env.device = MenderDevice(clients[0])
        env.device.ssh_is_opened()
```

Reply:
fixed. we not only return one device and expect one device.